### PR TITLE
Changed random chatter function

### PR DESCRIPTION
**Pull request by dennisOdenbreit**
-Added two more options
-Slightly changed the intervals of the other options
-Changed the code so that changing the random chatter selection is registered immediately rather than after the next random topic / 60 seconds on never
-Events before the random chat will be checked every so often.
-Tested and worked so far

https://github.com/Monika-After-Story/MonikaModDev/issues/3676

### CODE REVIEW

File: Monika After Story/game/definitions.rpy [hunk 1]
```diff
@@ -5546,7 +5555,64 @@ init 1 python in mas_randchat:
...
+       
+       
+    def startWaitProcess():
+        """
+        Use this method for convenience.
+        It can be used to actively wait the random chatter time.
+        """
+        startWaiting()
+        
+        while not waitedLongEnough():
```
Comment (ThePotatoGuy) [inline Monika After Story/game/definitions.rpy]: so if random chatter is set to NEVER, `waitedLongEnough` always returns False and this loop constantly waits(). This prevents idle operations in `ch30_loop` (script-ch30) from running (Event checking / delayedactions / seasonalchecks / filereactions / persistent saving / affection saving)

Adding a check in this function for the NEVER case where you wait, then jump to the `post_pick_random_topic` label could solve this issue.

Comment (dennisOdenbreit) [reply Monika After Story/game/definitions.rpy]: I suppose you mean the preloop needs to be checked constantly?

Comment (dennisOdenbreit) [reply Monika After Story/game/definitions.rpy]: I wasn't sure about this, I'll remove the while loop to keep it checked then.

Comment (ThePotatoGuy) [reply Monika After Story/game/definitions.rpy]: the `ch30_loop` label, down where it does a thing to `calendar_last_checked`. All of that is designed to run once per minute, but requires a jump to `ch3_loop` every so often.

Comment (ThePotatoGuy) [reply Monika After Story/game/definitions.rpy]: yeah I think removing the while loop might work if you instead jump to the `post_pick_random_topic` label if you haven't waited long enough.

**Pull request merged**